### PR TITLE
ref: fix typing in performance testutils

### PR DESCRIPTION
**Pull request by asottile-sentry**
mixins cannot be made typesafe unless they are strict subclasses so I moved the mixin method to a free function

best viewed with [?w=1](https://github.com/getsentry/sentry/pull/69711/files?w=1)

### CODE REVIEW

**Comment by vartec** [comment]
> mixins cannot be made typesafe unless they are strict subclasses

Isn't that possible using [Protocol](https://docs.python.org/3/library/typing.html#typing.Protocol)?

**Comment by asottile-sentry** [comment]
> > mixins cannot be made typesafe unless they are strict subclasses
> 
> Isn't that possible using [Protocol](https://docs.python.org/3/library/typing.html#typing.Protocol)?

not really -- because then you end up clobbering the mro with protocol methods.  and even if you get it to work at runtime nothing is validating that the mixins are safe and coherent with the protocol definitions

File: src/sentry/testutils/performance_issues/store_transaction.py [hunk 1]
```diff
@@ -4,71 +4,73 @@
 
 from django.utils import timezone
 
+from sentry.eventstore.models import Event
 from sentry.snuba.dataset import Dataset
 from sentry.testutils.cases import SnubaTestCase
+from sentry.utils import snuba
 
 
-class PerfIssueTransactionTestMixin:
```
Comment (armenzg) [inline src/sentry/testutils/performance_issues/store_transaction.py:11]: Oh cool! Burn it with 🔥 ! 😄

**Review by armenzg — APPROVED** [review]
This is great! Thanks @asottile-sentry !

**Pull request merged**